src/api/java/com/minecolonies/api/colony/buildings/registry/BuildingEntry.java

```diff
@@ -114,6 +117,18 @@ public Builder addBuildingModuleProducer(final Supplier<IBuildingModule> moduleP
             buildingModules.add(moduleProducer);
             return this;
         }
+
+        /**
+         * Add a building module producer.
+         * @param moduleProducer the module producer.
+         * @return the builder again.
+         */
+        public Builder addBuildingModuleProducer(final Supplier<IBuildingModule> moduleProducer, final Supplier<IBuildingModuleView> moduleViewProducer)
```

Comment:
You should add an overload here that only accepts the module provider and inserts a dummy moduleViewProvider.